skimage/transform/_geometric.py

```diff
@@ -230,13 +256,20 @@ class FundamentalMatrixTransform(GeometricTransform):
 
     """
 
-    def __init__(self, matrix=None):
+    def __init__(self, matrix=None, *, dimensionality=2):
         if matrix is None:
             # default to an identity transform
-            matrix = np.eye(3)
-        if matrix.shape != (3, 3):
+            matrix = np.eye(dimensionality + 1)
+        else:
+            dimensionality = matrix.shape[0] - 1
+        if matrix.shape != (dimensionality + 1, dimensionality + 1):
```

Comment:
Shouldn't `dimensionality` be deduced from `matrix.shape` if `matrix is not None`?
I mean, when matrix is provided, we can assume that the user knows the `dimensionality`, we don't need to force him to give it as argument if `dimensionality != 2`...
I think `dimensionality` should be `None` by default...

Reply:
Well, sorry, I need one more coffee this morning :sweat_smile:...

BTW, this situation can never happen since `dimensionality` is overwritten when `matrix is not None`.
```suggestion
```

Reply:
Waouh, I need a bucket of coffee...

Reply:
:joy: I actually checked the diff cover when developing tests here so I'm fairly confident that all lines are reachable. :smiley: